### DISCUSSION TITLE
Trying to figure out where the data is coming from... the Class?

### DISCUSSION THREAD

**Discussion by ferdware** [givanz/Vvveb#207] [General]
So in the "product.tpl" i was able to understand the following:

[data-v-test-test]|innerText = 'THIS IS A TEST'
and was able to display it on "product.html"

My question is in the product.tpl I'm trying to figure out what data is in the product model and where it comes from. 

I see "$this->product['name']" but I want to know what other values are in the product array.  Is see the var dump in the tpl but I get "Warning: Undefined property: Vvveb\System\Core\View::$product in....."

**Comment by givanz** [comment]
The product data is fetched in [app/component/product.php#L55](https://github.com/givanz/Vvveb/blob/master/app/component/product.php#L55) 

The product->get method from the product model is generated from  [admin/sql/mysqli/product.sql#L5](https://github.com/givanz/Vvveb/blob/master/admin/sql/mysqli/product.sql#L5) 

Properties are set from the component template with [app/template/components/product.tpl#L29](https://github.com/givanz/Vvveb/blob/master/app/template/components/product.tpl#L29)

You can `var_dump($results)` in request() method in product component to see all available properties [app/component/product.php#L102](https://github.com/givanz/Vvveb/blob/master/app/component/product.php#L102) the data from results() method is cached and is called only when the cache expires or is cleared from admin, the request() method is called on each request.

You can also check the compiled template at `vvveb/storage/compiled-templates/app_1_landing_product_product.html` to see what php code is generated from the template code.

Try `$product['name']` instead of `$this->product['name']` this is the variable set from the product component template to be used for `data-v-if` and other operations [app/template/components/product.tpl#L15](https://github.com/givanz/Vvveb/blob/master/app/template/components/product.tpl#L15) 

The data passed to view for product component can be accessed with `$this->_component['product'][0]` where 0 is the first product component on the page (some components can be inserted multiple times on the page with different options)

**Reply by ferdware** [reply]
Thank you so much!
